metomi/rose/suite_engine_procs/cylc.py

```diff
@@ -16,21 +16,28 @@
 # -----------------------------------------------------------------------------
 """Logic specific to the Cylc suite engine."""
 
-from glob import glob
 import os
 import pwd
 import re
-from random import shuffle
 import socket
 import sqlite3
 import tarfile
+
+from glob import glob
+from random import shuffle
+from typing import Union, List, Tuple, Optional, Iterable
 from time import sleep
 from uuid import uuid4
 
-from metomi.rose.fs_util import FileSystemEvent
-from metomi.rose.popen import RosePopenError
-from metomi.rose.reporter import Reporter
-from metomi.rose.suite_engine_proc import (
+from cylc.flow.platforms import get_host_from_platform
+from cylc.rose.platform_utils import (  # type:ignore
+    get_platforms_from_task_jobs
+)
+from cylc.rose.platform_utils import get_platform_from_task_def
```

Comment:
I think the best option, at least for now, is to put these imports inside the functions themselves to avoid import errors on this file.